api/metric/api.go

```diff
@@ -104,34 +104,60 @@ func (m Measurement) Number() core.Number {
 
 // Meter is an interface to the metrics portion of the OpenTelemetry SDK.
 type Meter interface {
```

Comment:
Breaking this apart into many "sub-interfaces" seems more OOP and less idiomatic Go. I'm not seeing any direct implementation of the constructor interfaces and only seeing them being used to encapsulate logically grouped method types. I'm not sure why there would ever by a type that implemented the constructors and not the `Meter` interface, making me wonder if this is a good approach.

Have I missed something?

Reply:
I was imagining a situation where an API needed to construct instruments but could declare itself as only using synchronous (or asynchronous) constructors for some reason.

Since you raised this point, I think we can go even further and just make the `MeterMust` struct support constructors only, no `Labels()` or `RecordBatch()` is needed. I've done this. I wonder if `MeterMust` should be renamed `MeterMustConstructors`, but I'm happy with it as-is now.